packages/react-dom/src/client/DOMPropertyOperations.js

```diff
@@ -110,7 +110,7 @@ export function getValueForAttribute(
       return expected === undefined ? undefined : null;
     }
     const value = node.getAttribute(name);
-    if (value === '' + (expected: any)) {
+    if (value === toString(expected)) {
```

Comment:
Why this change?

Reply:
if `expected` is the opaque object, this would throw the string error. `toString` doesn't convert the opaque object, which makes this just return `value`, which is what it should be anyway.